Simplify Working Group and OpenGitOps project Governance (#125)
gitops-working-group/gitops-working-group: GOVERNANCE.md

There are no special requirements for membership apart from the responsibilities listed  below.
Members are not required to have any explicit roles or additional responsibilities, but they may also have formally assigned roles (see below).
See the Working Group [README](README.md) for information about volunteering for committees and projects.
Note that voting requires additional conditions (see [Elections](#elections)).

Responsibilities:

- Must follow the [Code of Conduct](CODE_OF_CONDUCT.md)
- May not create the impression that they have any authority or formal responsibilities within the GitOps WG other than their explicitly assigned roles

### Committees

Working Group [Members](#all-members) may volunteer to join WG Committees and help with specific projects according to their interest and ability.
Responsibilities vary per committee, but in general please remember everyone is a volunteer.
Committees follow processes to allow for progress so long as they maintain inclusiveness, transparency, and respect for all involved.

Current WG committees:

- Principles Committee
  - Responsible for drafting the revised GitOps Principles (see [Timeline](#timeline))

> - GitOpsCon Committee
>   - Responsible for organizing GitOpsCon events (see [Timeline](#timeline))

Author: ```suggestion
- Events Committee
  - Responsible for organizing events such as GitOpsCon, and supporting other events in the wider GitOps community (see [Timeline](#timeline))
```

- Ensure discussion is extended asynchronously to be inclusive of members who cannot attend a specific meeting time
- Schedule discussion of proposals that have been submitted
- Partner with Maintainers and other WG members to establish a roadmap and manage ongoing projects
- Ask for new proposals to be made to address an identified needs
- Have GitOps domain knowledge
- Engage deeply in the work being done by the working group - that is, the role is not purely administrative.
  Chairs are, however not responsible for performing all of the work - this is shared across the working group members and expected from those who have volunteered onto various workstreams.
- When necessary, handle human and technical coordination across different working group workstreams
- Handle Code of Conduct violations
- Resolve escalated decisions when Maintainers can not reach consensus

Ultimately the chairs - after consulting with the collective of Maintainers and their community - drive the direction, values and governance of the overall project.

The GitHub `@gitops-working-group/chairs` team will be kept up to date with current chairs.

## Elections

> [Chairs](#chairs) will be elected for one year terms and may run for reelection.

Reviewer: Any thoughts about max # of consecutive terms for an individual?

Author: @csand-msft LIFETIME!!! 👑 lol j/k. No strong opinion on this one 🙂

My main concern is to keep this as simple as possible, while learning what we can from other CNCF project governance docs. Ultimately whatever's best for the stability, fairness, and momentum of the group and project.

Personally my gut feeling is setting max terms isn't necessary, since requiring reelection annually should self-govern and weed out anything weird. WDYT?

Author: We can also continue to iterate on governance after merging this PR.

The election process takes place on the CNCF SIG App Delivery [mailing list](cncf-sig-app-delivery@lists.cncf.io ).

> Before each election, there will be a call for nominiations on the mailing list.

Reviewer 2: ```suggestion
Before each election, there will be a call for nominations on the mailing list.
```

- Have GitOps domain knowledge
- Engage deeply in the work being done by the working group - that is, the role is not purely administrative.
  Chairs are, however not responsible for performing all of the work - this is shared across the working group members and expected from those who have volunteered onto various workstreams.
- When necessary, handle human and technical coordination across different working group workstreams

### Current Chairs

Name | Organization | GitHub
-- | -- | --
Dan Garfield | Codefresh | @todaywasawesome
Leonardo Murillo | Cloud Native Architects | @murillodigital
Scott Rigby | Weaveworks | @scottrigby

## Elections

[Chairs](#chairs) will be elected for one year terms and may run for reelection.

Chairs are selected by the members of the working group.

The election process takes place on the CNCF SIG App Delivery [mailing list](cncf-sig-app-delivery@lists.cncf.io ).

Before each election, there will be a call for nominiations on the mailing list.
Nominations should be sent to the mailing list, and include a brief bio and personal statement describing the candidates qualifications to serve in this capacity.
Self-nominations are welcome.

> Any member may cast a vote.

Author: This was my initial super-simple stab – assuming good will all around – though as written now it would be pretty easy to game the system if membership is self-declared and everyone can vote. We may want to do something similar to what CNF WG does, by having [interested parties](https://github.com/cncf/cnf-wg/blob/master/GOVERNANCE.md#elections) register within a certain time period before elections. I think our voting rules can be simpler though, since we aren't trying to balance 3 categorically different communities as they are. Thougths?

Author: I added an interested parties doc, with people and orgs listed in the initial GitOps WG charter.

# Governance

This document defines the governance process for the GitOps Working Group and community.

## Membership

> Membership is self-declared.

Reviewer 3: "While anyone may join and contribute, voting members must be active members of the CNCF. "

> Membership is self-declared.

Reviewer 3: ```suggestion
     While anyone may join and contribute, voting members must be active members of the CNCF.
```

A maximum of one person from any one entity may hold a chair role at any given moment in order to avoid undue influence, and maintain balanced representation.
[Current chairs](#current-chairs) are listed in this this document.

Responsibilities:

- Primary role of Chairs is governance of the Working Group, who collectively care for the OpenGitOps project
- Ensure that meetings and other activities are conducted and progress continues to be made against the project agenda, while also engaging other group members in leadership roles
- Ensure discussion is extended asynchronously to be inclusive of members who cannot attend a specific meeting time
- Schedule discussion of proposals that have been submitted
- Partner with other WG members to establish a roadmap and manage ongoing projects
- Ask for new proposals to be made to address an identified needs
- Have GitOps domain knowledge
- Engage deeply in the work being done by the working group - that is, the role is not purely administrative.
  Chairs are, however not responsible for performing all of the work - this is shared across the working group members and expected from those who have volunteered onto various workstreams.

> - When necessary, handle human and technical coordination across different working group workstreams

Reviewer 2: What do y'all think about also considering the definition of new initiatives as well as the promotion of open involvement as responsibilities for the chairs?

Author: > What do y'all think about also considering the definition of new initiatives 

I was thinking that's covered by this? WDYT
```
- Schedule discussion of proposals that have been submitted
- Partner with other WG members to establish a roadmap and manage ongoing projects
- Ask for new proposals to be made to address an identified needs
```

> as well as the promotion of open involvement as responsibilities for the chairs?

Maybe "Ensure the promotion of open involvement"? This way, it's not the sole responsibility of chairs to promote this, but only to ensure that it is in fact being promoted.

On this note, I had set up a "Communications" team in this GitHub org as an area to ask other interested people for help. For consistent messaging, I do think we need a dedicated team for this – however small – with good internal communication and clear processes. So yes, sgtm!

Author: @murillodigital  Added your second suggestion with commit co-authorship 👍  LMK if you disagree with the first point

### Current Chairs

Name | Organization | GitHub
-- | -- | --
Dan Garfield | Codefresh | @todaywasawesome
Leonardo Murillo | Cloud Native Architects | @murillodigital
Scott Rigby | Weaveworks | @scottrigby

## Elections

[Chairs](#chairs) will be elected for one year terms and may run for reelection.

The election process takes place on the CNCF SIG App Delivery [mailing list](cncf-sig-app-delivery@lists.cncf.io ).

Before each election, there will be a call for nominiations on the mailing list.
Nominations should be sent to the mailing list, and include a brief bio and personal statement describing the candidates qualifications to serve in this capacity.
Self-nominations are welcome.

The GitOps WG employs "organization voting" to ensure no single organization can dominate the election process or project.

> Each organization listed in the [interested parties](interested-parties.md) document has one vote.

Reviewer 3: "Up to two individuals from organizations who are both active members of the CNCF and listed in the [interested parties](interested-parties.md) may vote."

# Governance

This document defines the governance process for the GitOps Working Group and community.

## Membership

> While anyone may join and contribute, voting members must be active members of the CNCF.
> Apart from following the [Code of Conduct](CODE_OF_CONDUCT.md), there are no special requirements.
> This is an open public working group welcoming anyone who would like to help.
> Feel free to register your interest in the [interested parties](interested-parties.md) document.

Author: @todaywasawesome I think we should specify the voting requirements in only one place so I linked below instead. Then the "membership" wording needed to be switched around to still make sense. WDTY?

```suggestion
This is an open public working group welcoming anyone who would like to join and help.
Apart from following the [Code of Conduct](CODE_OF_CONDUCT.md), there are no special requirements for membership.
Feel free to register your interest in the [interested parties](interested-parties.md) document.
Note that voting requires additional conditions (see [Elections](#elections)).
```

If a vote is called, the following decisions require a Supermajority Vote <https://en.wikipedia.org/wiki/Supermajority>.

> - Chairs: Enforcing a Code of Conduct violation by a community member.

Reviewer 3: ```suggestion
- Maintainers: Adding additional maintainers.
- Chairs: Enforcing a Code of Conduct violation by a community member.
```
